pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -249,6 +251,10 @@ func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 			a.Type = "secret"
 			a.TypeChanged = true
 		}
+		if len(a.Type) == 0 && (len(a.ConfigMapName) > 0) {
```

Comment:
> It isn't entirely certain to me why if this is an add operation and a type and config map name are both provided, that the type should prevail.

typos.  If I specify conflicting values, I should fail either before or after I get here.  Now I don't actually see that logic ( :( ), but that's the reasoning behind this and the existing code.